### ISSUE TITLE
seems like http to https redirect is not working

### ISSUE THREAD

**Issue by may17** [cytopia/devilbox#265]
- operating system OSX
- docker version 18.03.0-ce
- docker-compose version version 1.20.1, build 5d8c71b

First of all thanks a lot for the new https feature.

I\`ve got one troubleshoot atm. It seems like the redirect from http to https is not working. I could open http://myproject.loc and https://myproject.loc and both are working. Shouldn`t been http://myproject.loc redirected to https://myproject.loc?

I am using HTTPD_SERVER nginx-stable with the default nginx template.

**Comment by cytopia** [comment]
The default is to have both, http and https.

A redirect can however be accomplished by setting:

```
      - MAIN_VHOST_SSL_TYPE=both
```
to
```
      - MAIN_VHOST_SSL_TYPE=redir
```
in docker-compose.yml

It is not yet a configurable variable in `.env` as it was not yet properly tested. Feel free to report back here how that works.

**Comment by may17** [comment]
Thank you for your fast response. I\`ve changed the MAIN_VHOST_SSL_TYPE to redir but it has no effect. The "problem" still exists.

If i changed it to redirect (Just to test if changes are reconized) the log shows me an error after running docker-compose up as expected.

`[ERR]  Invalid value for $MAIN_VHOST_SSL_TYPE: 'redirect'. Allowed: plain, ssl, both or redir`

**Comment by cytopia** [comment]
It should be `redir`, not `redirect`

**Comment by cytopia** [comment]
There are actually two places:
```
      - MAIN_VHOST_SSL_TYPE=redir
      - MASS_VHOST_SSL_TYPE=redir
```

**Comment by may17** [comment]
I`ve added redir to MAIN_VHOST_SSL_TYPE and MASS_VHOST_SSL_TYPE but it is still not redirecting.

**Comment by cytopia** [comment]
It does work fine for me. What is the output of docker logs after applying the above settings?

**Comment by cytopia** [comment]
@may17 is this still an issue for you?

**Comment by cytopia** [comment]
### This issue will be closed due to staleness.

Add required information in order to have it re-opened.

**Comment by iKlsR** [comment]
This isn't working for me either. Win 10 1903. Latest master.

**Comment by nicolabeghin** [comment]
I confirm the same issue, by setting both `MAIN_VHOST_SSL_TYPE` and `MASS_VHOST_SSL_TYPE` to `redir` no automatic redirect happens from HTTP to HTTPS. I'm on stock (updated) devilbox, apache-2.4 and php-fpm

**Comment by cytopia** [comment]
Re-opening, will have a closer look at [vhost-gen](https://github.com/devilbox/vhost-gen).

**Comment by nicolabeghin** [comment]
Any news on this? I would like to help, if some hint can be provided!

**Comment by cytopia** [comment]
It does work on the intranet with `MAIN_VHOST_SSL_TYPE=redir`. However it does not work on the projects via `MASS_VHOST_SSL_TYPE=redir`. I am still troubleshooting this.

**Comment by nicolabeghin** [comment]
saw the commit, great work @cytopia, kudos!!!

**Comment by cytopia** [comment]
Now fixed in [v1.4.0](https://github.com/cytopia/devilbox/releases/tag/v1.4.0)

**Comment by cytopia** [comment]
Now it is configurable via `.env`
